bin/knc.c

```diff
@@ -47,6 +55,10 @@
 #include <time.h>
 #include <unistd.h>
 
+#ifndef POLLRDHUP
+#define POLLRDHUP 0
```

Comment:
How about:

```
#ifndef POLRDHUP
#define POLLRDHUP POLLHUP
#endif
```

Reply:
POLLHUP should be ignored in pfd->events, but might theoretically result in EINVAL on some platforms.

Reply:
On Linux, man page suggests POLLHUP is ignored in events.  NetBSD man page documents that POLLHUP is checked even if not present in events, implying that it is accepted in events.  Ok.  I'll make that change, but also have to add a preprocessor check around setting the timeout for poll, since a timeout is suggested if POLLRDHUP is not supported.

Reply:
According to the spec, EWOULDBLOCK can be defined and different from EAGAIN, so I can not take that suggestion.  I'll add a check for EWOULDBLOCK for portability.

Reply:
Interesting -- the Open Group (POSIX, SUS) manpage says nothing about that.  The Linux one says that `POLLHUP` is meaningless in `events`.  Fine.

Reply:
Old reference to differences between poll() implementations on different unix and unix-like platforms: https://www.greenend.org.uk/rjk/tech/poll.html

Reply:
Please note that the reason that the existing code makes assumptions that it is called to move data only if 'work' is in a state that is ready to accept additional data.  I preserved that behavior to minimize the scope of code changes.  Therefore, I only check for POLLIN, which should only be returned if it was requested in pfd->events.  That's why I am not checking for (POLLIN|POLLRDHUP|POLLHUP) before read() so that read() detects EOF when the socket buffers are empty  Intead, I store a flag that POLLRDHUP|POLLHUP was received.